Scripts/Engine/CustomBar.lua

```diff
@@ -0,0 +1,848 @@
+	LevelVars.Engine.CustomBars.bars[dataName].textColor		= textColor
+	LevelVars.Engine.CustomBars.bars[dataName].hideText			= hideText  -- required to hide bar text
+	LevelVars.Engine.CustomBars.bars[dataName].visible			= false
+	LevelVars.Engine.CustomBars.bars[dataName].currentAlpha		= 0
+	LevelVars.Engine.CustomBars.bars[dataName].targetAlpha		= 0
+	LevelVars.Engine.CustomBars.bars[dataName].alphaBlendSpeed	= alphaBlendSpeed
+	LevelVars.Engine.CustomBars.bars[dataName].blink			= blink
+	LevelVars.Engine.CustomBars.bars[dataName].blinkLimit		= blinkLimit
+	LevelVars.Engine.CustomBars.bars[dataName].blinkSpeed		= 8
+	LevelVars.Engine.CustomBars.bars[dataName].showBar			= nil --required to hide bar when enemy is not targeted
+	LevelVars.Engine.CustomBars.bars[dataName].object			= nil
+	LevelVars.Engine.CustomBars.bars[dataName].getActionType	= nil
+	LevelVars.Engine.CustomBars.bars[dataName].currentTimer		= 0
+    return setmetatable(self, CustomBar)
+end
+
+---
+-- Creates a bar tied to Lara's attributes (Health, Air, Stamina).
+-- @tparam Objects.ObjID objectIDbg Object ID for the bar's background sprite.
+-- @tparam number spriteIDbg SpriteID from the specified object for the bar's background.
+-- @tparam Color colorbg Color of bar's background.
+-- @tparam Vec2 posBG X,Y position of the bar's background in screen percent (0-100).
+-- @tparam number rotBG rotation of the bar's background. sprite (0-360).
+-- @tparam Vec2 scaleBG X,Y Scaling factor for the bar's background sprite.
+-- @tparam View.AlignMode alignModebg Alignment for the bar's background.
+-- @tparam View.ScaleMode scaleModebg Scaling for the bar's background.
+-- @tparam Effects.BlendID blendModebg Blending modes for the bar's background.
+-- @tparam Objects.ObjID objectIDbar Object ID for the bar sprite.
+-- @tparam number spriteIDbar SpriteID from the specified object for the bar.
+-- @tparam Color colorbar Color of the bar.
+-- @tparam Vec2 posBar X,Y position of the bar in screen percent (0-100).
+-- @tparam number rot rotation of the bar's sprite (0-360).
+-- @tparam Vec2 scaleBar X,Y Scaling factor for the bar's sprite.
+-- @tparam View.AlignMode alignMode Alignment for the bar.
+-- @tparam View.ScaleMode scaleMode Scaling for the bar.
+-- @tparam Effects.BlendID blendMode Blending modes for the bar.
+-- @tparam number alphaBlendSpeed Speed of alpha blending for bar visibility (0-255).
+-- @tparam number getActionType Determines the bar type: 1: Health, 2: Air, 3: Stamina.
+-- @tparam bool showBar Option to always show the bar. If set to false, the bars will automatically hide when they stop updating.
+-- @tparam bool blink Whether the bar blinks.
+-- @tparam number blinkLimit % Limit below which bar starts blinking (0-1).
+--
+-- @treturn CustomBar The custombar in its hidden state
+CustomBar.CreateLaraBar = function (objectIDbg, spriteIDbg, colorbg, posBG, rotBG, scaleBG, alignModebg, scaleModebg, blendModebg,

```

Comment:
You can share same setup table structure with function above, just omitting unused fields.

Reply:
Done